File: MoveZeroes/move_zeroes.rb
def move_zeroes(nums)
    i = 0
    j = 0
    n = nums.length
    while j < n
        nums[i] = nums[j]
        j += 1
        i += 1 if nums[i] != 0
    end
    while i < n
        nums[i] = 0
        i += 1
    end
end

File: MoveZeroes/move_zeroes.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut i = 0;
        let mut j = 0;
        let n = nums.len();
        while j < n {
            nums[i] = nums[j];
            j += 1;
            if nums[i] != 0 {
                i += 1;
            }
        }
        while i < n {
            nums[i] = 0;
            i += 1;
        }
    }
}